### PR TITLE
Implement datasource switch (switch from feeder gateway to rpc full node)

### LINKED ISSUE #6: Implement datasource switch (switch from feeder gateway to rpc full node)
Description
------------------------
As Carbonable I should be able to Implement datasource switch (switch from feeder gateway to rpc full node)

You can improve command line arguments to add flag either to use feeder gateway or full node as datasource 

Additional Information
---------------------------
[High Difficulty  - Go - Medium Priority]

Acceptance Criteria
-------------------------
- [ ] Implement datasource switch

### PR DESCRIPTION
**Pull request by joaquinbarrientoscr**
Resolves: #6 
- Added `cli_utils` package to have modularity when working with cli flags.
- Added `DatasourceBlock` interface to manage instances of datasources. Either `Feeder Gateway` or `Full Node RPC`

### CODE REVIEW

File: internal/cli_utils/datasource_flag.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package cli_utils
+
+import (
+	"flag"
+	"fmt"
+	"slices"
+)
+
+type datasource_flag struct {
```
Comment (MartianGreed) [inline internal/cli_utils/datasource_flag.go]: type name should follow go conventions : DatasourceFlag

File: internal/cli_utils/datasource_flag.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	Value    string
+	Is_valid bool
+}
+
+func (d *datasource_flag) datasource_binding() {
```
Comment (MartianGreed) [inline internal/cli_utils/datasource_flag.go]: func name should be datasouceBinding (if private) and DatasourceBinding (if public)

File: internal/cli_utils/datasource_flag.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	flag.StringVar(&d.Value, "datasource", "fg", "")
+	flag.StringVar(&d.Value, "d", "fg", "")
+}
+
+func (d *datasource_flag) Validation() {
+	/*
+		fg = feed gateway
+		fn = full node rpc
+	*/
```
Comment (MartianGreed) [inline internal/cli_utils/datasource_flag.go]: if guess we should extract those into a const like so : 
```
type Datasource string
const (
	FeederGateway  Datasource = "feeder_gateway"
	FullNode Datasource = "full_node"
)

```

File: internal/cli_utils/datasource_flag.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	valid_values := []string{"fg", "fn"}
```
Comment (MartianGreed) [inline internal/cli_utils/datasource_flag.go]: therefore we can select case on those two consts to validate user input

File: internal/datasource/block_datasource.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package datasource
```
Comment (MartianGreed) [inline internal/datasource/block_datasource.go]: I guess this is supposed to go the synchronizer package

File: internal/datasource/feeder_gateway.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package datasource
```
Comment (MartianGreed) [inline internal/datasource/feeder_gateway.go]: I guess this is supposed to go the synchronizer package

File: internal/datasource/full_node.go [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package datasource
```
Comment (MartianGreed) [inline internal/datasource/full_node.go]: I guess this is supposed to go the synchronizer package

**Review by MartianGreed — CHANGES_REQUESTED** [review]

File: internal/cli_utils/datasource_flag.go (hunk 1 above)
Comment (joaquinbarrientoscr) [reply internal/cli_utils/datasource_flag.go]: Done

File: internal/cli_utils/datasource_flag.go (hunk 3 above)
Comment (joaquinbarrientoscr) [reply internal/cli_utils/datasource_flag.go]: Done, thanks!

File: internal/cli_utils/datasource_flag.go (hunk 4 above)
Comment (joaquinbarrientoscr) [reply internal/cli_utils/datasource_flag.go]: sure!
as discussed via Telegram, a _SWITCH CASE_ was used for this scenario

File: internal/datasource/block_datasource.go (hunk 5 above)
Comment (joaquinbarrientoscr) [reply internal/datasource/block_datasource.go]: Done

**Review by MartianGreed — APPROVED** [review]

**Pull request merged**